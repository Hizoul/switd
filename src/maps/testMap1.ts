import Map from "../logic/map"
import Street, { direction } from "../logic/street"

const testMap1 = new Map()

const start = new Street(0, 0, testMap1)
testMap1.addTile(start)
const s1 = start.extend(direction.right)
const s2 = s1.extend(direction.down)
const s31 = s2.extend(direction.down)
const s32 = s2.extend(direction.right)
const s321 = s31.extend(direction.right)
const s322 = s321.extend(direction.down)
const s311 = s31.extend(direction.down)
const s312 = s311.extend(direction.right)
const s4intersection = s312.extend(direction.right)
s322.next.push(s4intersection)
s4intersection.previous.push(s322)

const s5 = s4intersection.extend(direction.down)
const s6 = s5.extend(direction.down)
const s7 = s6.extend(direction.left)
const s8 = s7.extend(direction.left)

export default testMap1
