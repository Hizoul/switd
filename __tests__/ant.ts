import Ant from "../src/logic/ant"
import testMap1 from "../src/maps/testMap1"

const matchAntState = (ant: Ant, msg: string) => {
  expect({
    x: ant.currentlyOn.xPos,
    y: ant.currentlyOn.yPos,
    isStart: ant.currentlyOn.isStart(),
    isTarget: ant.currentlyOn.isTarget()
  }).toMatchSnapshot(msg)
}

it(`Ants can walk a path`, () => {
  const ant1 = new Ant(testMap1.getStart())
  matchAntState(ant1, "currently at start")
  ant1.makeNextStep()
  matchAntState(ant1, "currently at first pos")
  ant1.makeNextStep()
  matchAntState(ant1, "currently at second pos")
  ant1.makeNextStep()
  matchAntState(ant1, "currently at third pos")
})
