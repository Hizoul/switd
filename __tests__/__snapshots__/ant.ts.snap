// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currently at first pos 1`] = `
Object {
  "isStart": false,
  "isTarget": false,
  "x": 1,
  "y": 0,
}
`;

exports[`currently at second pos 1`] = `
Object {
  "isStart": false,
  "isTarget": false,
  "x": 1,
  "y": -1,
}
`;

exports[`currently at start 1`] = `
Object {
  "isStart": true,
  "isTarget": false,
  "x": 0,
  "y": 0,
}
`;

exports[`currently at third pos 1`] = `
Object {
  "isStart": false,
  "isTarget": true,
  "x": 2,
  "y": -1,
}
`;
